### ISSUE TITLE
Customizable Loading state on QTable Component

### ISSUE THREAD

**Issue by datadisc** [quasarframework/quasar#5965]
**Problem.**
Let's say that we have a "high" table (100 rows). In every row, along with date, we have buttons which executest "long lasting" actions for the record on the server side. Natural solutions is to switch the table into the loading state. The first problem is that the indicator is located under the header and is not always visible when you scroll the table (not sticky header variant ). The second problem is that the indicator does not block the content of the table which in some cases could be desirable behaviour.

**Desirable solution**
QTable could optionally use `QInnerLoading` (examples: [Buefy](https://buefy.org/documentation/table), [Kendo](https://www.telerik.com/kendo-vue-ui/components/grid-native/)) to block whole table / table body. Another solution would be additional slot as a child of the `q-table__container`. In this slot we could easily put `QInnerLoading` component and use it instead of `QTable`'s loading state.

**Current solution**
The action buttons on each row, based on loading state, could be blocked individually (I think this is the "expensive" solution).

Another solution is to wrap the table in another div and put the `QInnerLoading` as a child of this div but it does not work in the fullscreen mode - the `QInnerLoading` is simply "hidden" below the QTable.

**Comment by metalsadman** [comment]
> Another solution is to wrap the table in another div and put the `QInnerLoading` as a child of this div but it does not work in the fullscreen mode - the `QInnerLoading` is simply "hidden" below the QTable.

use class `z-top` class on your q-innerloader.

**Comment by hawkeye64** [comment]
@datadisc You may be wanting something like this app-extension: https://quasarframework.github.io/quasar-ui-qoverlay/examples

**Comment by rstoenescu** [comment]
New scoped slot ("loading") will be available in "quasar" v1.8.
Example:

```html
  <q-table :loading="loading" .... >
      <template v-slot:loading>
        <q-inner-loading showing color="primary" />
      </template>
    </q-table>
```

**Comment by acetilholin** [comment]
Is it possible to adjust speed of loading bar?
